In [601]:
import numpy as np
import pandas as pd

In [640]:
data = pd.read_csv("auto_complectations.csv")

In [641]:
data.head()

brand     model               complect        type_car  seats  doors  \
0  Audi  A3 Sedan          1.4 TFSI 7AMT  Седан класса C      5      4   
1  Audi  A3 Sedan    sport 1.4 TFSI 7AMT  Седан класса C      5      4   
2  Audi  A3 Sedan          2.0 TFSI 7AMT  Седан класса C      5      4   
3  Audi  A3 Sedan  2.0 TFSI 7AMT quattro  Седан класса C      5      4   
4  Audi  A3 Sedan    sport 2.0 TFSI 7AMT  Седан класса C      5      4   

               Двигатель   Мощность Крутящий момент двигателя  \
0  бензиновый (1395 см³)  150 л. с.                   250 Н·м   
1  бензиновый (1395 см³)  150 л. с.                   250 Н·м   
2  бензиновый (1984 см³)  190 л. с.                   320 Н·м   
3  бензиновый (1984 см³)  190 л. с.                   320 Н·м   
4  бензиновый (1984 см³)  190 л. с.                   320 Н·м   

                 Коробка передач  ... Разгон до сотни Максимальная скорость  \
0  роботизированная (7 ступеней)  ...     8.2 секунды              224 км/ч   
1  роботизированная (7 ступеней)  ...     8.2 секунды              224 км/ч   
2  роботизированная (7 ступеней)  ...     6.8 секунды              250 км/ч   
3  роботизированная (7 ступеней)  ...     6.2 секунды              242 км/ч   
4  роботизированная (7 ступеней)  ...     6.8 секунды              242 км/ч   

  Расход топлива (л/100 км)\nгород / трасса / смешанный Дорожный просвет  \
0                                    5.9 / 4.1 / 4.8              165 мм   
1                                    5.9 / 4.1 / 4.8              165 мм   
2                                    7.2 / 4.7 / 5.6              165 мм   
3                                    7.2 / 4.8 / 5.7              165 мм   
4                                    7.2 / 4.7 / 5.6              165 мм   

  Габариты (длина × ширина × высота) Колёсная база Объём багажника  \
0                 4458 × 1796 × 1416       2637 мм           425 л   
1                 4458 × 1796 × 1416       2637 мм           425 л   
2                 4458 × 1796 × 1416       2637 мм           425 л   
3                 4458 × 1796 × 1416       2637 мм           390 л   
4                 4458 × 1796 × 1416       2637 мм           425 л   

  Объём багажника максимальный Объём топливного бака Масса автомобиля  
0                        880 л                  50 л          1320 кг  
1                        880 л                  50 л          1320 кг  
2                        880 л                  50 л          1395 кг  
3                        845 л                  55 л          1465 кг  
4                        880 л                  50 л          1395 кг  

[5 rows x 21 columns]

In [642]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1703 entries, 0 to 1702
Data columns (total 21 columns):
 #   Column                                                Non-Null Count  Dtype 
---  ------                                                --------------  ----- 
 0   brand                                                 1703 non-null   object
 1   model                                                 1703 non-null   object
 2   complect                                              1703 non-null   object
 3   type_car                                              1703 non-null   object
 4   seats                                                 1703 non-null   int64 
 5   doors                                                 1703 non-null   int64 
 6   Двигатель                                             1703 non-null   object
 7   Мощность                                              1703 non-null   object
 8   Крутящий момент двигателя                             1703 non-null 

### Смотрим на уникальные значения

In [643]:
def show_unique_data(data):
    for i in range(data.shape[1]):
        print('column name: ', data.columns[i]) 
        print('number of unique: ', data[data.columns[i]].unique().size) 
        print(data[data.columns[i]].unique())
        print()

In [644]:
show_unique_data(data)

column name:  brand
number of unique:  27
['Audi' 'BMW' 'Cadillac' 'Chevrolet' 'Citroen' 'Genesis' 'Infiniti'
 'Hyundai' 'Honda' 'Jaguar' 'Jeep' 'KIA' 'Land Rover' 'Lexus' 'Mazda'
 'Mercedes-Benz' 'Mitsubishi' 'Nissan' 'Opel' 'Peugeot' 'Renault' 'Skoda'
 'Subaru' 'Suzuki' 'Toyota' 'Volkswagen' 'Volvo']

column name:  model
number of unique:  246
['A3 Sedan' 'A3 Sportback' 'A4' 'A4 Allroad' 'A4 Avant' 'A5'
 'A5 Sportback' 'A6' 'A6 Avant' 'A7 Sportback' 'A8' 'e-tron' 'Q3'
 'Q3 Sportback' 'Q5' 'Q7' 'Q8' 'S6' 'S7 Sportback' 'S8' 'SQ5' 'SQ7' 'SQ8'
 '2 Series' '2 Series Gran Coupe' '3 GT' '3 Series Sedan' '4 Series'
 '5 Series Sedan (2016)' '5 Series Sedan' '6 GT' '7 Series Sedan'
 '8 Series' '8 Series Cabrio' '8 Series Gran Coupe' 'i3' 'I8' 'i8 Cabrio'
 'M2' 'M4' 'M5' 'M8 Cabrio' 'M8 Coupe' 'M8 Gran Coupe' 'X1' 'X2' 'X3'
 'X3 M' 'X4' 'X4 M' 'X5' 'X5 M' 'X6' 'X6 M' 'X7' 'Z4 Roadster' 'CT6'
 'Escalade' 'XT5' 'XT6' 'Cobalt' 'Nexia' 'Spark' 'Tahoe' 'Traverse'
 'C3 Aircross' 'C4 Sedan' 'C5 Aircr

In [645]:
# Вывод всех данных (если без break)
for i, row in data.iterrows():
    print(f"Index: {i}")
    print(f"{row}\n")
    break

Index: 0
brand                                                                             Audi
model                                                                         A3 Sedan
complect                                                                 1.4 TFSI 7AMT
type_car                                                                Седан класса C
seats                                                                                5
doors                                                                                4
Двигатель                                                        бензиновый (1395 см³)
Мощность                                                                     150 л. с.
Крутящий момент двигателя                                                      250 Н·м
Коробка передач                                          роботизированная (7 ступеней)
Привод                                                                        передний
Разгон до сотни                   

### Смотрим, где есть пропуски в данных

In [646]:
def no_info_in_data(data):
    for col_name, dt in data.items():
        for i in range(data.shape[0]):
            if dt[i] == 'нет данных':
                #print(data.iloc[i])
                print(i, col_name)

In [647]:
no_info_in_data(data)

342 Разгон до сотни
344 Разгон до сотни
346 Разгон до сотни
575 Разгон до сотни
576 Разгон до сотни
577 Разгон до сотни
587 Разгон до сотни
588 Разгон до сотни
589 Разгон до сотни
590 Разгон до сотни
591 Разгон до сотни
592 Разгон до сотни
947 Разгон до сотни
1094 Разгон до сотни
1095 Разгон до сотни
1096 Разгон до сотни
1097 Разгон до сотни
1098 Разгон до сотни
1388 Разгон до сотни
1389 Разгон до сотни
1390 Разгон до сотни
1429 Разгон до сотни
1430 Разгон до сотни
1431 Разгон до сотни
1432 Разгон до сотни
1433 Разгон до сотни
1434 Разгон до сотни
1435 Разгон до сотни
1436 Разгон до сотни
1437 Разгон до сотни
1438 Разгон до сотни
1456 Разгон до сотни
1458 Разгон до сотни
1460 Разгон до сотни
1462 Разгон до сотни
1464 Разгон до сотни
1466 Разгон до сотни
1535 Разгон до сотни
1536 Разгон до сотни
1537 Разгон до сотни
1538 Разгон до сотни
1539 Разгон до сотни
1540 Разгон до сотни
1541 Разгон до сотни
1542 Разгон до сотни
1543 Разгон до сотни
1544 Разгон до сотни
1545 Разгон до сотни
1546 

Видим, что только в столбце "Разгон до сотни" нет данных. 

Заменим значения 'нет данных' на нули

Будем генерить фичи, преобразовывать переменные. 

## Преобразование данных (object -> float)

In [648]:
data['seast'] = data['seats'].astype('float64')
data['doors'] = data['doors'].astype('float64')

## Преобразование данных (Разбиение столбцов)

### Двигатель

In [649]:
temp = data['Двигатель'].apply(lambda x: str.split(x, ' '))
del data['Двигатель']
temp

0       [бензиновый, (1395, см³)]
1       [бензиновый, (1395, см³)]
2       [бензиновый, (1984, см³)]
3       [бензиновый, (1984, см³)]
4       [бензиновый, (1984, см³)]
                  ...            
1698     [дизельный, (1969, см³)]
1699    [бензиновый, (1969, см³)]
1700     [дизельный, (1969, см³)]
1701     [гибридный, (1969, см³)]
1702     [гибридный, (1969, см³)]
Name: Двигатель, Length: 1703, dtype: object

In [650]:
data['тип_двигателя'] = temp.apply(lambda x: x[0])
data['тип_двигателя']

0       бензиновый
1       бензиновый
2       бензиновый
3       бензиновый
4       бензиновый
           ...    
1698     дизельный
1699    бензиновый
1700     дизельный
1701     гибридный
1702     гибридный
Name: тип_двигателя, Length: 1703, dtype: object

In [651]:
data['объем_двигателя'] = temp.apply(lambda x: x[1]).apply(lambda y: str.split(y, '(')).apply(lambda x: x[1]).astype('float64')
data['объем_двигателя']

0       1395.0
1       1395.0
2       1984.0
3       1984.0
4       1984.0
         ...  
1698    1969.0
1699    1969.0
1700    1969.0
1701    1969.0
1702    1969.0
Name: объем_двигателя, Length: 1703, dtype: float64

### Мощность

In [652]:
temp = data['Мощность'].apply(lambda x: str.split(x, ' '))
del data['Мощность']
data['мощность_двигателя'] = temp.apply(lambda x: x[0]).astype('float64')
data['мощность_двигателя']

0       150.0
1       150.0
2       190.0
3       190.0
4       190.0
        ...  
1698    235.0
1699    320.0
1700    235.0
1701    407.0
1702    407.0
Name: мощность_двигателя, Length: 1703, dtype: float64

### Крутящий момент двигателя

In [653]:
temp = data['Крутящий момент двигателя'].apply(lambda x: str.split(x, ' '))
del data['Крутящий момент двигателя']
data['крутящий_момент_двигателя'] = temp.apply(lambda x: x[0]).astype('float64')
data['крутящий_момент_двигателя']

0       250.0
1       250.0
2       320.0
3       320.0
4       320.0
        ...  
1698    480.0
1699    400.0
1700    480.0
1701    640.0
1702    640.0
Name: крутящий_момент_двигателя, Length: 1703, dtype: float64

### Коробка передач

In [654]:
temp = data['Коробка передач'].apply(lambda x: str.split(x, ' '))
del data['Коробка передач']
data['тип_коробки_передач'] = temp.apply(lambda x: x[0])
data['тип_коробки_передач'] 

0       роботизированная
1       роботизированная
2       роботизированная
3       роботизированная
4       роботизированная
              ...       
1698      автоматическая
1699      автоматическая
1700      автоматическая
1701      автоматическая
1702      автоматическая
Name: тип_коробки_передач, Length: 1703, dtype: object

In [655]:
data['количество_ступеней'] = temp.apply(lambda x: x[1]).apply(lambda y: str.split(y, '(')).apply(lambda x: x[1]).astype('float64')
data['количество_ступеней']

0       7.0
1       7.0
2       7.0
3       7.0
4       7.0
       ... 
1698    8.0
1699    8.0
1700    8.0
1701    8.0
1702    8.0
Name: количество_ступеней, Length: 1703, dtype: float64

### Разгон до сотни

In [656]:
data.loc[data['Разгон до сотни'] == 'нет данных', 'Разгон до сотни'] = '0'

In [657]:
temp = data['Разгон до сотни'].apply(lambda x: str.split(x, ' '))
del data['Разгон до сотни']
data['разгон_до_сотни'] = temp.apply(lambda x: x[0]).astype('float64')
data['разгон_до_сотни']

0       8.2
1       8.2
2       6.8
3       6.2
4       6.8
       ... 
1698    7.8
1699    6.5
1700    7.8
1701    5.6
1702    5.6
Name: разгон_до_сотни, Length: 1703, dtype: float64

 ### Максимальная скорость

In [658]:
temp = data['Максимальная скорость'].apply(lambda x: str.split(x, ' '))
del data['Максимальная скорость']
data['максимальная_скорость'] = temp.apply(lambda x: x[0]).astype('float64')
data['максимальная_скорость']

0       224.0
1       224.0
2       250.0
3       242.0
4       242.0
        ...  
1698    220.0
1699    230.0
1700    220.0
1701    230.0
1702    230.0
Name: максимальная_скорость, Length: 1703, dtype: float64

### Расход топлива

In [659]:
temp = data['Расход топлива (л/100 км)\nгород / трасса / смешанный'].apply(lambda x: str.split(x, ' / '))
del data['Расход топлива (л/100 км)\nгород / трасса / смешанный']
print(temp)
data['расход_топлива_город'] = temp.apply(lambda x: x[0]).astype('float64')
data['расход_топлива_трасса'] = temp.apply(lambda x: x[1]).astype('float64')
data['расход_топлива_смешанный'] = temp.apply(lambda x: x[2]).astype('float64')

0       [5.9, 4.1, 4.8]
1       [5.9, 4.1, 4.8]
2       [7.2, 4.7, 5.6]
3       [7.2, 4.8, 5.7]
4       [7.2, 4.7, 5.6]
             ...       
1698    [6.4, 5.5, 5.8]
1699        [0, 0, 8.3]
1700    [6.4, 5.5, 5.8]
1701        [0, 0, 2.1]
1702        [0, 0, 2.1]
Name: Расход топлива (л/100 км)\nгород / трасса / смешанный, Length: 1703, dtype: object


### Дорожный просвет

In [660]:
temp = data['Дорожный просвет'].apply(lambda x: str.split(x, ' '))
del data['Дорожный просвет']
data['дорожный_просвет'] = temp.apply(lambda x: x[0]).astype('float64')
data['дорожный_просвет']

0       165.0
1       165.0
2       165.0
3       165.0
4       165.0
        ...  
1698    237.0
1699    237.0
1700    237.0
1701    237.0
1702    237.0
Name: дорожный_просвет, Length: 1703, dtype: float64

### Габариты

In [661]:
temp = data['Габариты (длина × ширина × высота)'].apply(lambda x: str.split(x, ' × '))
del data['Габариты (длина × ширина × высота)']
print(temp)
data['длина_машины'] = temp.apply(lambda x: x[0]).astype('float64')
data['ширина_машины'] = temp.apply(lambda x: x[1]).astype('float64')
data['высота_машины'] = temp.apply(lambda x: x[2]).astype('float64')

0       [4458, 1796, 1416]
1       [4458, 1796, 1416]
2       [4458, 1796, 1416]
3       [4458, 1796, 1416]
4       [4458, 1796, 1416]
               ...        
1698    [4950, 2008, 1776]
1699    [4950, 2008, 1776]
1700    [4950, 2008, 1776]
1701    [4950, 2008, 1776]
1702    [4950, 2008, 1776]
Name: Габариты (длина × ширина × высота), Length: 1703, dtype: object


### Колесная база

In [662]:
temp = data['Колёсная база'].apply(lambda x: str.split(x, ' '))
del data['Колёсная база']
data['колёсная_база'] = temp.apply(lambda x: x[0]).astype('float64')
data['колёсная_база']

0       2637.0
1       2637.0
2       2637.0
3       2637.0
4       2637.0
         ...  
1698    2984.0
1699    2984.0
1700    2984.0
1701    2984.0
1702    2984.0
Name: колёсная_база, Length: 1703, dtype: float64

### Объем багажника

In [663]:
temp = data['Объём багажника'].apply(lambda x: str.split(x, ' '))
del data['Объём багажника']
data['объем_багажника'] = temp.apply(lambda x: x[0]).astype('float64')
data['объем_багажника']

0       425.0
1       425.0
2       425.0
3       390.0
4       425.0
        ...  
1698    721.0
1699    692.0
1700    692.0
1701    640.0
1702    640.0
Name: объем_багажника, Length: 1703, dtype: float64

### Объем багажника максимальный

In [664]:
temp = data['Объём багажника максимальный'].apply(lambda x: str.split(x, ' '))
del data['Объём багажника максимальный']
data['объём_багажника_максимальный'] = temp.apply(lambda x: x[0]).astype('float64')
data['объём_багажника_максимальный']

0        880.0
1        880.0
2        880.0
3        845.0
4        880.0
         ...  
1698    1899.0
1699    1899.0
1700    1899.0
1701    1899.0
1702    1899.0
Name: объём_багажника_максимальный, Length: 1703, dtype: float64

### Объем топливного бака

In [665]:
temp = data['Объём топливного бака'].apply(lambda x: str.split(x, ' '))
del data['Объём топливного бака']
data['объём_топливного_бака'] = temp.apply(lambda x: x[0]).astype('float64')
data['объём_топливного_бака']

0       50.0
1       50.0
2       50.0
3       55.0
4       50.0
        ... 
1698    71.0
1699    71.0
1700    71.0
1701    70.0
1702    70.0
Name: объём_топливного_бака, Length: 1703, dtype: float64

### Масса автомобиля

In [666]:
temp = data['Масса автомобиля'].apply(lambda x: str.split(x, ' '))
del data['Масса автомобиля']
data['масса_автомобиля'] = temp.apply(lambda x: x[0]).astype('float64')
data['масса_автомобиля']

0       1320.0
1       1320.0
2       1395.0
3       1465.0
4       1395.0
         ...  
1698    1969.0
1699    2004.0
1700    2009.0
1701    2350.0
1702    2350.0
Name: масса_автомобиля, Length: 1703, dtype: float64

In [667]:
data

brand     model               complect           type_car  seats  doors  \
0      Audi  A3 Sedan          1.4 TFSI 7AMT     Седан класса C      5    4.0   
1      Audi  A3 Sedan    sport 1.4 TFSI 7AMT     Седан класса C      5    4.0   
2      Audi  A3 Sedan          2.0 TFSI 7AMT     Седан класса C      5    4.0   
3      Audi  A3 Sedan  2.0 TFSI 7AMT quattro     Седан класса C      5    4.0   
4      Audi  A3 Sedan    sport 2.0 TFSI 7AMT     Седан класса C      5    4.0   
...     ...       ...                    ...                ...    ...    ...   
1698  Volvo      XC90      D5 AT 5S R-Design  Большой кроссовер      5    5.0   
1699  Volvo      XC90      T6 AT 7S R-Design  Большой кроссовер      7    5.0   
1700  Volvo      XC90      D5 AT 7S R-Design  Большой кроссовер      7    5.0   
1701  Volvo      XC90   T8 AT 7S Inscription  Большой кроссовер      7    5.0   
1702  Volvo      XC90      T8 AT 7S R-Design  Большой кроссовер      7    5.0   

        Привод  seast тип_двигателя  объем_двигателя  ...  \
0     передний    5.0    бензиновый           1395.0  ...   
1     передний    5.0    бензиновый           1395.0  ...   
2     передний    5.0    бензиновый           1984.0  ...   
3       полный    5.0    бензиновый           1984.0  ...   
4     передний    5.0    бензиновый           1984.0  ...   
...        ...    ...           ...              ...  ...   
1698    полный    5.0     дизельный           1969.0  ...   
1699    полный    7.0    бензиновый           1969.0  ...   
1700    полный    7.0     дизельный           1969.0  ...   
1701    полный    7.0     гибридный           1969.0  ...   
1702    полный    7.0     гибридный           1969.0  ...   

      расход_топлива_смешанный  дорожный_просвет длина_машины  ширина_машины  \
0                          4.8             165.0       4458.0         1796.0   
1                          4.8             165.0       4458.0         1796.0   
2                          5.6             165.0       4458.0         1796.0   
3                          5.7             165.0       4458.0         1796.0   
4                          5.6             165.0       4458.0         1796.0   
...                        ...               ...          ...            ...   
1698                       5.8             237.0       4950.0         2008.0   
1699                       8.3             237.0       4950.0         2008.0   
1700                       5.8             237.0       4950.0         2008.0   
1701                       2.1             237.0       4950.0         2008.0   
1702                       2.1             237.0       4950.0         2008.0   

      высота_машины  колёсная_база  объем_багажника  \
0            1416.0         2637.0            425.0   
1            1416.0         2637.0            425.0   
2            1416.0         2637.0            425.0   
3            1416.0         2637.0            390.0   
4            1416.0         2637.0            425.0   
...             ...            ...              ...   
1698         1776.0         2984.0            721.0   
1699         1776.0         2984.0            692.0   
1700         1776.0         2984.0            692.0   
1701         1776.0         2984.0            640.0   
1702         1776.0         2984.0            640.0   

      объём_багажника_максимальный  объём_топливного_бака  масса_автомобиля  
0                            880.0                   50.0            1320.0  
1                            880.0                   50.0            1320.0  
2                            880.0                   50.0            1395.0  
3                            845.0                   55.0            1465.0  
4                            880.0                   50.0            1395.0  
...                            ...                    ...               ...  
1698                        1899.0                   71.0            1969.0  
1699                        1899.0                   7

## Преобразование данных (работа с категориальными данными)

### Привод

In [668]:
data['Привод'] = str('привод_') + data['Привод']
data = pd.concat([data, pd.get_dummies(data['Привод'])], axis=1).drop('Привод', axis=1)
data.head()

brand     model               complect        type_car  seats  doors  seast  \
0  Audi  A3 Sedan          1.4 TFSI 7AMT  Седан класса C      5    4.0    5.0   
1  Audi  A3 Sedan    sport 1.4 TFSI 7AMT  Седан класса C      5    4.0    5.0   
2  Audi  A3 Sedan          2.0 TFSI 7AMT  Седан класса C      5    4.0    5.0   
3  Audi  A3 Sedan  2.0 TFSI 7AMT quattro  Седан класса C      5    4.0    5.0   
4  Audi  A3 Sedan    sport 2.0 TFSI 7AMT  Седан класса C      5    4.0    5.0   

  тип_двигателя  объем_двигателя  мощность_двигателя  ...  ширина_машины  \
0    бензиновый           1395.0               150.0  ...         1796.0   
1    бензиновый           1395.0               150.0  ...         1796.0   
2    бензиновый           1984.0               190.0  ...         1796.0   
3    бензиновый           1984.0               190.0  ...         1796.0   
4    бензиновый           1984.0               190.0  ...         1796.0   

  высота_машины  колёсная_база  объем_багажника  объём_багажника_максимальный  \
0        1416.0         2637.0            425.0                         880.0   
1        1416.0         2637.0            425.0                         880.0   
2        1416.0         2637.0            425.0                         880.0   
3        1416.0         2637.0            390.0                         845.0   
4        1416.0         2637.0            425.0                         880.0   

   объём_топливного_бака  масса_автомобиля  привод_задний  привод_передний  \
0                   50.0            1320.0              0                1   
1                   50.0            1320.0              0                1   
2                   50.0            1395.0              0                1   
3                   55.0            1465.0              0                0   
4                   50.0            1395.0              0                1   

   привод_полный  
0              0  
1              0  
2              0  
3              1  
4              0  

[5 rows x 30 columns]

### Комплектация

Уникальных значений 1400, всего значений 1703

Удаляем столбец, т.к. он скорее ухудшит работу алгоритма, чем улчучшит

In [669]:
del data['complect']

### Модель

In [670]:
data['model'] = str('model ') + data['model']
data = pd.concat([data, pd.get_dummies(data['model'])], axis=1).drop('model', axis=1)
data.head()

brand        type_car  seats  doors  seast тип_двигателя  объем_двигателя  \
0  Audi  Седан класса C      5    4.0    5.0    бензиновый           1395.0   
1  Audi  Седан класса C      5    4.0    5.0    бензиновый           1395.0   
2  Audi  Седан класса C      5    4.0    5.0    бензиновый           1984.0   
3  Audi  Седан класса C      5    4.0    5.0    бензиновый           1984.0   
4  Audi  Седан класса C      5    4.0    5.0    бензиновый           1984.0   

   мощность_двигателя  крутящий_момент_двигателя тип_коробки_передач  ...  \
0               150.0                      250.0    роботизированная  ...   
1               150.0                      250.0    роботизированная  ...   
2               190.0                      320.0    роботизированная  ...   
3               190.0                      320.0    роботизированная  ...   
4               190.0                      320.0    роботизированная  ...   

   model XJ  model XT5  model XT6  model XV  model Z4 Roadster  \
0         0          0          0         0                  0   
1         0          0          0         0                  0   
2         0          0          0         0                  0   
3         0          0          0         0                  0   
4         0          0          0         0                  0   

   model Zafira Life  model e-tron  model i3  model i30 N  model i8 Cabrio  
0                  0             0         0            0                0  
1                  0             0         0            0                0  
2                  0             0         0            0                0  
3                  0             0         0            0                0  
4                  0             0         0            0                0  

[5 rows x 274 columns]

### Бренд

In [671]:
data['brand'] = str('brand ') + data['brand']
data = pd.concat([data, pd.get_dummies(data['brand'])], axis=1).drop('brand', axis=1)
data.head()

type_car  seats  doors  seast тип_двигателя  объем_двигателя  \
0  Седан класса C      5    4.0    5.0    бензиновый           1395.0   
1  Седан класса C      5    4.0    5.0    бензиновый           1395.0   
2  Седан класса C      5    4.0    5.0    бензиновый           1984.0   
3  Седан класса C      5    4.0    5.0    бензиновый           1984.0   
4  Седан класса C      5    4.0    5.0    бензиновый           1984.0   

   мощность_двигателя  крутящий_момент_двигателя тип_коробки_передач  \
0               150.0                      250.0    роботизированная   
1               150.0                      250.0    роботизированная   
2               190.0                      320.0    роботизированная   
3               190.0                      320.0    роботизированная   
4               190.0                      320.0    роботизированная   

   количество_ступеней  ...  brand Nissan  brand Opel  brand Peugeot  \
0                  7.0  ...             0           0              0   
1                  7.0  ...             0           0              0   
2                  7.0  ...             0           0              0   
3                  7.0  ...             0           0              0   
4                  7.0  ...             0           0              0   

   brand Renault  brand Skoda  brand Subaru  brand Suzuki  brand Toyota  \
0              0            0             0             0             0   
1              0            0             0             0             0   
2              0            0             0             0             0   
3              0            0             0             0             0   
4              0            0             0             0             0   

   brand Volkswagen  brand Volvo  
0                 0            0  
1                 0            0  
2                 0            0  
3                 0            0  
4                 0            0  

[5 rows x 300 columns]

### Тип двигателя

In [672]:
data['тип_двигателя'] = str('тип_двигателя ') + data['тип_двигателя']
data = pd.concat([data, pd.get_dummies(data['тип_двигателя'])], axis=1).drop('тип_двигателя', axis=1)
data.head()

type_car  seats  doors  seast  объем_двигателя  мощность_двигателя  \
0  Седан класса C      5    4.0    5.0           1395.0               150.0   
1  Седан класса C      5    4.0    5.0           1395.0               150.0   
2  Седан класса C      5    4.0    5.0           1984.0               190.0   
3  Седан класса C      5    4.0    5.0           1984.0               190.0   
4  Седан класса C      5    4.0    5.0           1984.0               190.0   

   крутящий_момент_двигателя тип_коробки_передач  количество_ступеней  \
0                      250.0    роботизированная                  7.0   
1                      250.0    роботизированная                  7.0   
2                      320.0    роботизированная                  7.0   
3                      320.0    роботизированная                  7.0   
4                      320.0    роботизированная                  7.0   

   разгон_до_сотни  ...  brand Skoda  brand Subaru  brand Suzuki  \
0              8.2  ...            0             0             0   
1              8.2  ...            0             0             0   
2              6.8  ...            0             0             0   
3              6.2  ...            0             0             0   
4              6.8  ...            0             0             0   

   brand Toyota  brand Volkswagen  brand Volvo  тип_двигателя бензиновый  \
0             0                 0            0                         1   
1             0                 0            0                         1   
2             0                 0            0                         1   
3             0                 0            0                         1   
4             0                 0            0                         1   

   тип_двигателя гибридный  тип_двигателя дизельный  \
0                        0                        0   
1                        0                        0   
2                        0                        0   
3                        0                        0   
4                        0                        0   

   тип_двигателя электрический  
0                            0  
1                            0  
2                            0  
3                            0  
4                            0  

[5 rows x 303 columns]

### Тип коробки передач

In [673]:
data['тип_коробки_передач'] = str('тип_коробки_передач ') + data['тип_коробки_передач']
data = pd.concat([data, pd.get_dummies(data['тип_коробки_передач'])], axis=1).drop('тип_коробки_передач', axis=1)
data.head()

type_car  seats  doors  seast  объем_двигателя  мощность_двигателя  \
0  Седан класса C      5    4.0    5.0           1395.0               150.0   
1  Седан класса C      5    4.0    5.0           1395.0               150.0   
2  Седан класса C      5    4.0    5.0           1984.0               190.0   
3  Седан класса C      5    4.0    5.0           1984.0               190.0   
4  Седан класса C      5    4.0    5.0           1984.0               190.0   

   крутящий_момент_двигателя  количество_ступеней  разгон_до_сотни  \
0                      250.0                  7.0              8.2   
1                      250.0                  7.0              8.2   
2                      320.0                  7.0              6.8   
3                      320.0                  7.0              6.2   
4                      320.0                  7.0              6.8   

   максимальная_скорость  ...  brand Volkswagen  brand Volvo  \
0                  224.0  ...                 0            0   
1                  224.0  ...                 0            0   
2                  250.0  ...                 0            0   
3                  242.0  ...                 0            0   
4                  242.0  ...                 0            0   

   тип_двигателя бензиновый  тип_двигателя гибридный  тип_двигателя дизельный  \
0                         1                        0                        0   
1                         1                        0                        0   
2                         1                        0                        0   
3                         1                        0                        0   
4                         1                        0                        0   

   тип_двигателя электрический  тип_коробки_передач автоматическая  \
0                            0                                   0   
1                            0                                   0   
2                            0                                   0   
3                            0                                   0   
4                            0                                   0   

   тип_коробки_передач вариатор  тип_коробки_передач механическая  \
0                             0                                 0   
1                             0                                 0   
2                             0                                 0   
3                             0                                 0   
4                             0                                 0   

   тип_коробки_передач роботизированная  
0                                     1  
1                                     1  
2                                     1  
3                                     1  
4                                     1  

[5 rows x 306 columns]

## Сохраняем датасет

In [677]:
data.to_csv('update_auto_complectations.csv', index=False)